lsp/src/text_document.ml

```diff
@@ -155,9 +154,10 @@ let apply_content_change ?version (t : t)
       in
       { t.document with text }
   in
-  let document =
-    match version with
-    | None -> document
-    | Some version -> { document with version }
-  in
   { t with document }
+
+let apply_content_changes ?version t changes =
+  let t = List.fold_left apply_change t changes in
+  match version with
+  | None -> t
```

Comment:
I don't see where we're incrementing the version? 

```suggestion
  | None ->
    { t with document = { t.document with version = t.document.version + 1 } }
```

Reply:
I don't think it's needed. The spec says the versions are provided by the client to the server.